hr_timesheet_time_restriction/tests/test_disabled_hr_timesheet_time_restriction.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 Dinar Gabbasov
+# Copyright 2022 Ooops404
+# Copyright 2022 Cetmix
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import datetime
+
+from odoo.exceptions import ValidationError
+from odoo.tests import common
+
+
+class TestHrTimesheetTimeRestriction(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestHrTimesheetTimeRestriction, cls).setUpClass()
+        # We create an active employee for the current user, if he is not yet
+        employee = cls.env["hr.employee"].search(
+            [("user_id", "=", cls.env.uid)], limit=1
+        )
+        if not employee:
+            employee = cls.env["hr.employee"].create(
+                {
+                    "name": cls.env.user.name or "Test Employee",
+                    "user_id": cls.env.uid,
+                    "active": True,
+                    "company_id": cls.env.company.id,
+                }
+            )
+        cls.employee = employee
+
+        # We create a project, task and configuration
+        cls.project = cls.env["project.project"].create({"name": "Test project"})
+        cls.analytic_account = cls.project.analytic_account_id
+        cls.task = cls.env["project.task"].create(
+            {
+                "name": "Test task",
+                "project_id": cls.project.id,
+            }
+        )
+        cls.config = cls.env["res.config.settings"].create({})
+        # The global restriction is disabled by default
+        cls.config.use_timesheet_restriction = False
+        cls.config.execute()
+
+    def test_project_restriction_days(self):
+        """With a disconnected global restriction (use_timesheet_restriction = False),
+        the restriction specified at the project level is not applied
+        """
+        self.project.timesheet_restriction_days = 1
+        # check that we can create new timesheet
+        line = self.env["account.analytic.line"].create(
+            {
+                "date": datetime.date.today(),
+                "task_id": self.task.id,
+                "project_id": self.project.id,
+                "account_id": self.analytic_account.id,
+                "employee_id": self.employee.id,
+                "name": "Test line",
+            }
+        )
+        self.assertTrue(line, "Timesheet should be created for today")
+        # check that we can create new timesheet with date before
+        # that current date - 1
+        line = self.env["account.analytic.line"].create(
```

Comment:
Duplicate code fragment with lines 128-137, 168-177. Consider pulling it out to setUp or setUpClass depending to business logic.